app/models/tube_rack.rb

```diff
@@ -78,4 +90,14 @@ def after_comment_addition(comment)
     # tubes that don't have submissions. Or even a mixed rack.
     comments.add_comment_to_tubes(comment)
   end
+
+  # Returns a hash of tube locations in the tube rack.
+  # The hash keys are the coordinates of the racked tubes, and the values are hashes containing the UUIDs of the tubes.
+  #
+  # @return [Hash] A hash of tube locations with coordinates as keys and tube UUIDs as values.
+  def tube_locations
```

Comment:
It is OK to have this on the model but I have noticed the writer tube_locations= is on the resource. Do you think we should move this to the resource?